pkg/istio-agent/health/health_probers.go

```diff
@@ -54,6 +58,49 @@ func (p *ProbeResult) IsHealthy() bool {
 	return *p == Healthy
 }
 
+type GRPCProber struct {
+	Config *v1alpha3.GrpcHealthCheckConfig
+}
+
+var _ Prober = &GRPCProber{}
+
+func NewGRPCProber(cfg *v1alpha3.GrpcHealthCheckConfig) *GRPCProber {
+	return &GRPCProber{
+		Config: cfg,
+	}
+}
+
+func (g *GRPCProber) Probe(timeout time.Duration) (ProbeResult, error) {
+	if g.Config == nil {
+		return Unknown, fmt.Errorf("grpc health check config is nil")
+	}
+	conn, err := grpc.NewClient(net.JoinHostPort("localhost", fmt.Sprintf("%d", g.Config.Port)),
```

Comment:
I would prefer to use grpc.DialContext. We use this in 100% of the rest of the code, and it allows us to more clearly see we are behaving the same way as Kubernete's probe

Reply:
```
  60   │         grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
  61   │             return probe.ProbeDialer().DialContext(ctx, "tcp", addr)
  62   │         }),
```

we will want this and `grpc.WithBlock(),`. See https://github.com/kubernetes/kubernetes/blob/master/pkg/probe/grpc/grpc.go. Our goal is to match as close as possible their behavior

Reply:
Got it these methods were marked deprecated in favour of `grpc.NewClient` and this didn't support options like `WithBlock()`. Will use it regardless to match K8s.